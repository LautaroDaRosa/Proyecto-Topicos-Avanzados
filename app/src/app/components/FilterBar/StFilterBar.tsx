import styled from 'styled-components';

const StFilterBar = styled.div`
  background: white;
  border-radius: 8px;
  padding: 24px;
  box-shadow: 0px 0px 20px 0px lightgrey;
  width: 100%;
  display: flex;
  justify-content: space-between;

  input {
    height: 40px;
    border-radius: 8px;
    border: 1px solid lightgray;
    padding-left: 12px;
    padding-right: 12px;
  }

  select {
    height: 40px;
    border-radius: 8px;
    border: 1px solid lightgray;
    padding-left: 12px;
    padding-right: 12px;
    cursor: pointer;
  }

  option {
    font-weight: normal;
    cursor: pointer;
    text-align: center;
  }

  option:checked {
    font-weight: bold;
    cursor: pointer;
    background-color: white;
  }

  button {
    height: 40px;
  }
`;

export default StFilterBar;
