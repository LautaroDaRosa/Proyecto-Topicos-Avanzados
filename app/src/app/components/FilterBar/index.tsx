import { useState } from 'react';
import Button from '../Button';
import StFilterBar from './StFilterBar';

interface FilterBarProps {
  onSearch: (
    name: string,
    businessName: string,
    rut: string,
    score: string,
    category: string,
  ) => void;
  clearFilters: () => void;
  categories: string[];
}

const FilterBar = ({ onSearch, clearFilters, categories }: FilterBarProps) => {
  const [name, setName] = useState('');
  const [businessName, setBusinessName] = useState('');
  const [rut, setRut] = useState('');
  const [category, setCategory] = useState('');
  const [score, setScore] = useState('');

  const scores = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

  const clearAllFilters = () => {
    setName('');
    setBusinessName('');
    setRut('');
    setCategory('');
    setScore('');
    clearFilters();
  };

  const emptyFilters =
    name === '' &&
    businessName === '' &&
    rut === '' &&
    category === '' &&
    score === '';

  const handleSearch = () => {
    onSearch(name, businessName, rut, score, category);
  };

  return (
    <StFilterBar>
      <input
        placeholder="Nombre"
        value={name}
        onChange={e => setName(e.target.value)}
      ></input>
      <input
        placeholder="Razon social"
        value={businessName}
        onChange={e => setBusinessName(e.target.value)}
      ></input>
      <input
        placeholder="RUT"
        value={rut}
        onChange={e => setRut(e.target.value)}
      ></input>
      <select value={score} onChange={e => setScore(e.target.value)}>
        <option value="" disabled>
          Score
        </option>
        {scores.map(score => (
          <option value={score} key={score}>
            {score}
          </option>
        ))}
      </select>

      <select value={category} onChange={e => setCategory(e.target.value)}>
        <option value="" disabled>
          Rubro
        </option>
        {categories.map(cat => (
          <option value={cat} key={cat}>
            {cat}
          </option>
        ))}
      </select>
      <div style={{ gap: '8px', display: 'flex' }}>
        <Button
          text="Limpiar"
          type="submit"
          action="secondary"
          onClick={clearAllFilters}
        />
        <Button
          text="Buscar"
          type="submit"
          action="primary"
          onClick={handleSearch}
          disabled={emptyFilters}
        />
      </div>
    </StFilterBar>
  );
};

export default FilterBar;
