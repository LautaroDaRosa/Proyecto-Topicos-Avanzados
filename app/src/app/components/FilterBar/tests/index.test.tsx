import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';

import FilterBar from '..';

describe('Filter Bar', () => {
  it('should match snapshot', () => {
    let x = 0;
    const handleSearch = (
      name: string,
      businessName: string,
      rut: string,
      score: string,
      category: string,
    ) => {
      x++;
    };

    const clearFilters = () => {
      x++;
    };
    const filterBar = render(
      <FilterBar
        clearFilters={clearFilters}
        onSearch={handleSearch}
        categories={['Cat 1', 'Cat 2']}
      />,
    );
    expect(filterBar).toMatchSnapshot();
    expect(x === 0);
  });
  it('should call the clearFilters function when the clear button is pressed.', () => {
    let x = 0;
    const handleSearch = (
      name: string,
      businessName: string,
      rut: string,
      score: string,
      category: string,
    ) => {
      x = 2;
    };

    const clearFilters = () => {
      x = 1;
    };
    const filterBar = render(
      <FilterBar
        clearFilters={clearFilters}
        onSearch={handleSearch}
        categories={['Cat 1', 'Cat 2']}
      />,
    );

    const buttonClear = filterBar.getByText('Limpiar');
    expect(x === 0);
    fireEvent.click(buttonClear);
    expect(x === 1);
  });

  it('should disable the search button if the user does not write anything', () => {
    let x = 0;
    const handleSearch = (
      name: string,
      businessName: string,
      rut: string,
      score: string,
      category: string,
    ) => {
      x = 2;
    };

    const clearFilters = () => {
      x = 1;
    };
    const filterBar = render(
      <FilterBar
        clearFilters={clearFilters}
        onSearch={handleSearch}
        categories={['Cat 1', 'Cat 2']}
      />,
    );

    const buttonSearch = filterBar.getByText('Buscar');
    expect(buttonSearch).toBeDisabled();
    expect(x === 0);
  });
  it('should enable the search button and call handleSearch if the user writes something in one input', () => {
    let x = 0;
    const handleSearch = (
      name: string,
      businessName: string,
      rut: string,
      score: string,
      category: string,
    ) => {
      x = 2;
    };

    const clearFilters = () => {
      x = 1;
    };
    const filterBar = render(
      <FilterBar
        clearFilters={clearFilters}
        onSearch={handleSearch}
        categories={['Cat 1', 'Cat 2']}
      />,
    );

    const buttonSearch = filterBar.getByText('Buscar');
    expect(buttonSearch).toBeDisabled();
    expect(x === 0);

    const nameInput = filterBar.getByPlaceholderText('Nombre');
    fireEvent.change(nameInput, { target: { value: 'Canal' } });
    expect(buttonSearch).not.toBeDisabled();
    expect(x === 2);
  });
});
