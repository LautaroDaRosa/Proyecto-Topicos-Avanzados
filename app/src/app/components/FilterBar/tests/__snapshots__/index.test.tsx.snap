// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Bar should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  border-radius: 8px;
  height: 48px;
  font-size: 16px;
  padding: 12px 24px;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  border: 1.5px solid #04b0f4;
  background-color: white;
  color: #04b0f4;
}

.c1:disabled {
  border-color: #c0bfcb;
  color: #c0bfcb;
}

.c1:hover:enabled {
  cursor: pointer;
  color: white;
  background-color: #04b0f4;
}

.c1:active:enabled {
  color: white;
  background-color: #137ca5;
  border-color: #137ca5;
}

.c2 {
  border-radius: 8px;
  height: 48px;
  font-size: 16px;
  padding: 12px 24px;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  border: 1.5px solid #fca408;
  background-color: white;
  color: #fca408;
}

.c2:disabled {
  border-color: #c0bfcb;
  color: #c0bfcb;
}

.c2:hover:enabled {
  cursor: pointer;
  color: white;
  background-color: #fca408;
}

.c2:active:enabled {
  color: white;
  background-color: #b27508;
  border-color: #b27508;
}

.c0 {
  background: white;
  border-radius: 8px;
  padding: 24px;
  box-shadow: 0px 0px 20px 0px lightgrey;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 input {
  height: 40px;
  border-radius: 8px;
  border: 1px solid lightgray;
  padding-left: 12px;
  padding-right: 12px;
}

.c0 select {
  height: 40px;
  border-radius: 8px;
  border: 1px solid lightgray;
  padding-left: 12px;
  padding-right: 12px;
  cursor: pointer;
}

.c0 option {
  font-weight: normal;
  cursor: pointer;
  text-align: center;
}

.c0 option:checked {
  font-weight: bold;
  cursor: pointer;
  background-color: white;
}

.c0 button {
  height: 40px;
}

<body>
    <div>
      <div
        class="c0"
      >
        <input
          placeholder="Nombre"
          value=""
        />
        <input
          placeholder="Razon social"
          value=""
        />
        <input
          placeholder="RUT"
          value=""
        />
        <select>
          <option
            disabled=""
            value=""
          >
            Score
          </option>
          <option
            value="1"
          >
            1
          </option>
          <option
            value="2"
          >
            2
          </option>
          <option
            value="3"
          >
            3
          </option>
          <option
            value="4"
          >
            4
          </option>
          <option
            value="5"
          >
            5
          </option>
          <option
            value="6"
          >
            6
          </option>
          <option
            value="7"
          >
            7
          </option>
          <option
            value="8"
          >
            8
          </option>
          <option
            value="9"
          >
            9
          </option>
          <option
            value="10"
          >
            10
          </option>
        </select>
        <select>
          <option
            disabled=""
            value=""
          >
            Rubro
          </option>
          <option
            value="Cat 1"
          >
            Cat 1
          </option>
          <option
            value="Cat 2"
          >
            Cat 2
          </option>
        </select>
        <div
          style="gap: 8px; display: flex;"
        >
          <button
            action="secondary"
            class="c1"
            type="submit"
          >
            Limpiar
          </button>
          <button
            action="primary"
            class="c2"
            disabled=""
            type="submit"
          >
            Buscar
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": .c1 {
  border-radius: 8px;
  height: 48px;
  font-size: 16px;
  padding: 12px 24px;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  border: 1.5px solid #04b0f4;
  background-color: white;
  color: #04b0f4;
}

.c1:disabled {
  border-color: #c0bfcb;
  color: #c0bfcb;
}

.c1:hover:enabled {
  cursor: pointer;
  color: white;
  background-color: #04b0f4;
}

.c1:active:enabled {
  color: white;
  background-color: #137ca5;
  border-color: #137ca5;
}

.c2 {
  border-radius: 8px;
  height: 48px;
  font-size: 16px;
  padding: 12px 24px;
  -webkit-transition: 0.1s;
  transition: 0.1s;
  border: 1.5px solid #fca408;
  background-color: white;
  color: #fca408;
}

.c2:disabled {
  border-color: #c0bfcb;
  color: #c0bfcb;
}

.c2:hover:enabled {
  cursor: pointer;
  color: white;
  background-color: #fca408;
}

.c2:active:enabled {
  color: white;
  background-color: #b27508;
  border-color: #b27508;
}

.c0 {
  background: white;
  border-radius: 8px;
  padding: 24px;
  box-shadow: 0px 0px 20px 0px lightgrey;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 input {
  height: 40px;
  border-radius: 8px;
  border: 1px solid lightgray;
  padding-left: 12px;
  padding-right: 12px;
}

.c0 select {
  height: 40px;
  border-radius: 8px;
  border: 1px solid lightgray;
  padding-left: 12px;
  padding-right: 12px;
  cursor: pointer;
}

.c0 option {
  font-weight: normal;
  cursor: pointer;
  text-align: center;
}

.c0 option:checked {
  font-weight: bold;
  cursor: pointer;
  background-color: white;
}

.c0 button {
  height: 40px;
}

<div>
    <div
      class="c0"
    >
      <input
        placeholder="Nombre"
        value=""
      />
      <input
        placeholder="Razon social"
        value=""
      />
      <input
        placeholder="RUT"
        value=""
      />
      <select>
        <option
          disabled=""
          value=""
        >
          Score
        </option>
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="3"
        >
          3
        </option>
        <option
          value="4"
        >
          4
        </option>
        <option
          value="5"
        >
          5
        </option>
        <option
          value="6"
        >
          6
        </option>
        <option
          value="7"
        >
          7
        </option>
        <option
          value="8"
        >
          8
        </option>
        <option
          value="9"
        >
          9
        </option>
        <option
          value="10"
        >
          10
        </option>
      </select>
      <select>
        <option
          disabled=""
          value=""
        >
          Rubro
        </option>
        <option
          value="Cat 1"
        >
          Cat 1
        </option>
        <option
          value="Cat 2"
        >
          Cat 2
        </option>
      </select>
      <div
        style="gap: 8px; display: flex;"
      >
        <button
          action="secondary"
          class="c1"
          type="submit"
        >
          Limpiar
        </button>
        <button
          action="primary"
          class="c2"
          disabled=""
          type="submit"
        >
          Buscar
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
